crates/sui/src/client_commands.rs

```diff
@@ -1658,35 +1661,123 @@ fn convert_number_to_string(value: Value) -> Value {
+//     writeln!(writer, "{}", response.digest)?;
+//     writeln!(writer, "{}", "----- Transaction Data ----".bold())?;
+//     if let Some(t) = response.transaction {
+//         writeln!(writer, "{}", t)?;
+//     }
+
+//     writeln!(writer, "{}", "----- Transaction Effects ----".bold())?;
+//     if let Some(e) = response.effects {
+//         writeln!(writer, "{}", e)?;
+//     }
+
+//     writeln!(writer, "{}", "----- Events ----".bold())?;
+//     if let Some(e) = response.events {
+//         writeln!(writer, "{:#?}", json!(e))?;
+//     }
+
+//     writeln!(writer, "{}", "----- Object changes ----".bold())?;
+//     if let Some(e) = response.object_changes {
+//         writeln!(writer, "{:#?}", json!(e))?;
+//     }
+
+//     writeln!(writer, "{}", "----- Balance changes ----".bold())?;
+//     if let Some(e) = response.balance_changes {
+//         writeln!(writer, "{:#?}", json!(e))?;
+//     }
+//     Ok(writer)
+// }
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionEffectsOutput {
+    status: String,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    created: Vec<OwnedObjectRef>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    mutated: Vec<OwnedObjectRef>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    deleted: Vec<SuiObjectRef>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    wrapped: Vec<SuiObjectRef>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    unwrapped: Vec<OwnedObjectRef>,
+}
+
+impl From<&SuiTransactionBlockEffects> for TransactionEffectsOutput {
+    fn from(tx_effects: &SuiTransactionBlockEffects) -> Self {
+        Self {
+            status: match tx_effects.status() {
+                SuiExecutionStatus::Success => "success".to_string(),
+                SuiExecutionStatus::Failure { error } => format!("error: {error}"),
+            },
+            created: tx_effects.created().to_vec(),
+            mutated: tx_effects.mutated().to_vec(),
+            deleted: tx_effects.deleted().to_vec(),
+            wrapped: tx_effects.wrapped().to_vec(),
+            unwrapped: tx_effects.unwrapped().to_vec(),
+        }
+    }
+}
+
+pub fn convert_tx_response_to_table(
     response: &SuiTransactionBlockResponse,
 ) -> Result<String, fmt::Error> {
+    let style = TableStyle::rounded().horizontals([]);
     let mut writer = String::new();
     writeln!(writer, "{}", "----- Transaction Digest ----".bold())?;
     writeln!(writer, "{}", response.digest)?;
     writeln!(writer, "{}", "----- Transaction Data ----".bold())?;
     if let Some(t) = &response.transaction {
-        writeln!(writer, "{}", t)?;
+        let json_obj = json!(t.data);
+        let mut table = json_to_table(&json_obj);
+        table.with(style.clone());
+        writeln!(writer, "{}", table)?;
+    }
+    writeln!(writer, "{}", "----- Transaction Signatures ----".bold())?;
+    if let Some(t) = &response.transaction {
```

Comment:
I am 100% with you on this one. I guess by default the `tabled` and `json-to-table` crates treat `Vec` as a new table with X rows, and therefore they display it with its own borders. There is one `collapse` function that I can try to see if it looks better, I will report back.

Reply:
Re-reading what you said, I misunderstood earlier. So, in theory, we could take care of these special cases with vectors of length one, but it would require some manual effort. I am considering leaving it as is for now before I figure out better ways to handle this.

Reply:
Could you handle this by writing a function that modifies the JSON? I.e. if it sees an array of length one it replaces it with its content?